src/NeoSharp.Core/Persistence/IRepositoryOfT.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Linq.Expressions;
+
+namespace NeoSharp.Core.Persistence
+{
+    public interface IRepository<TEntity> 
+        where TEntity : NeoEntityBase
+    {
+        void Create(TEntity entity);
```

Comment:
Could we have async versions as well?

Reply:
good idea.. going to create the structure of that

Reply:
I would even say the async version has to be the only one.

Reply:
agree!!!